### ISSUE TITLE
[Feature request] Copy GIF animations to clipboard

### ISSUE THREAD

**Issue by EirikBirkeland** [NickeManarin/ScreenToGif#251]
I'd like to be able to quickly copy/paste animations into Github -- currently, the clipboard content seems to contain an image only.

As Nicke explained to me, this may not be possible to implement under current framework/OS constraints.

Maybe a super-long-term goal?

**Comment by ashleedawg** [comment]
@EirikBirkeland  - I believe what Nick would have meant is that what you're asking isn't possible because of [**how Windows Clipboard is designed**](https://msdn.microsoft.com/en-us/library/windows/desktop/ff729168(v=vs.85).aspx).  Without getting into an explanation of *why,* trust me: there are a number of reasons that it is very unlikely that Microsoft would ever make the changes necessary to provide support for what you're asking. 

I must admit I'm a curious _where_ you are copying from (ie., online vs. specific app vs. Windows filesystem), and _how often_ you're doing it, that makes a couple clicks an inconvenience?  Perhaps you're not aware of how easy it already is to insert an Animated GIF from elsewhere (without copy/paste):

To copy an Animated GIF from a website:
-
1. Right-click the image and click "Save Image As" (or your browser's equivalent), saving it somewhere on your computer.
2. In **`Screen2Gif`**, hit <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>O</kbd> **or** click **Media** under **File** menu.. 
 3. Choose the image from the location on your computer where you saved it.

To insert a copy of an Animated GIF from a file:
-
 - Start at Step 2 above. 

---

Even if what you're suggesting can be done, it would take a ton of work and redesign, which isn't a reasonable expectation from a one-man team providing a free piece of awesome software -- especially for such negligible user benefit (to occasionally save a couple clicks). Alternatively, *this is an open project* if you want to hire a developer...  :wink: 
---
**_Side note:_** the  <sub>![bug](https://imgur.com/x5qeZNo.gif)</sub> label that you used is incorrect in this case. A bug would be when the software produces an incorrect or unexpected result, or to behave in unintended ways contrary to how the developer intended it to operate.  I think a more accurate label would be <sub>![Future Feature](https://imgur.com/qVY6Xbx.gif)</sub>.

**Comment by TomasHubelbauer** [comment]
Excuse my ignorance, but couldn't you, upon clipboard keystroke, generate a temporary file with the animation and then copy that as a file rather than as a bitmap so when the user goes to insert something into the GitHub editor, they have just a generic file in the clipboard? I am not sure how GitHub deals with the past if it's a file (maybe it turns it into an attachment), but otherwise I think this could work.

**Comment by EirikBirkeland** [comment]
@ashleedawg Nick labelled it a bug, not me. It don't remember my use case, but if it's unrealistic just close the issue. And stop attributing actions to me that I haven't carried out. And drop the smugness, it has no place in this forum.

tl;dr; your post is too long and ranty to address, but good luck doing something more useful with your time.

**Comment by trustserve** [comment]
Currently you can copy a link to clipboard for imagr. would love to see ability to copy richtext embed of that link to allow pasting image into any richtext window like gmail but others as well. i know screencastomatic does this to very good effect. its a much beloved feature that this does not have.

**Comment by dozd** [comment]
This is the feature I am really missing. I am using windows `snip & sketch` to share portions of my screen a lot and I would like same workflow with ScreenToGif.

Just fast capture my screen, ctrl+c and share on whenever I want. I know I can save and than share, but that one step is really downgrade in terms of efficiency ;).

**Comment by NickeManarin** [comment]
> Just fast capture my screen, ctrl+c and share on whenever I want. I know I can save and than share, but that one step is really downgrade in terms of efficiency ;).

Not possible and not my fault :P
If apps behaved like Telegram (which is able to load files copied from Windows Explorer), that would be possible.

**Comment by dozd** [comment]
It's like you cannot create a temp file or something and put it in clipboard?

**Comment by NickeManarin** [comment]
The issue is with the Windows API. 

To be able to paste something like in Discord or even in here, you need to "copy as image". Literally `Clipboard.SetImage()`.
The issue is that the method only supports BMP/PNG images.

Gif frames are lost on the conversion (that happens inside `SetImage()`), only the first frame being pasted.

**Comment by snorga** [comment]
![image](https://user-images.githubusercontent.com/909778/130454969-55d38c0c-8a64-4a20-a085-d0c1b986f6df.png)

This seems to do it?

I make the recording, do a "Save as" (write to a temp file), and then from the notification in top, right click and select "copy". 

So its posible to do from screenToGit app only.. ish

**Comment by popbee** [comment]
The menu option @snorga  is talking about works great in `Slack`.

What about options to `autosave after recording` and `autocopy file after saving` to reduce the number of steps for quick clips.

---
> If apps behaved like Telegram (which is able to load files copied from Windows Explorer), that would be possible.

`Slack` desktop app support proper image embedding when copying a file (tested on windows). It's awesome!
`Teams` desktop app (windows) does not seem to.  `Jira` (web) neither (I didn't try all jira editors there are a few).  `Discord` (web) does not.


> would love to see ability to copy richtext embed of that link to allow pasting image into any richtext window like gmail but others as well. i know **screencastomatic** does this to very good effect

This tool uploads the file "in the cloud" and creates a **public link** and puts that in the clipboard. That's good only in certain cases. It's not so great for my use cases. Also it requires some cloud solution for storage. How long will the image be kept?

> right-clicking one of the GIF results and choosing "Copy image". So it is possible, you just gotta figure out how.

@TheYarin : **Nope.** This only copies the first image, not the whole gif. Exactly like the @NickeManarin is saying. So please find another "working" example because so far, I have none that works universally.
